dynamiq/nodes/agents/base.py

```diff
@@ -71,10 +78,12 @@ def __init__(self, **kwargs):
         self._intermediate_steps: dict[int, dict] = {}
         self._run_depends: list[dict] = []
         self._init_prompt_blocks()
+        if self.memory is None:
+            self.memory = Memory()
```

Comment:
I think memory should be optional. Now we use it always. What if the user don't want to use memory for some reason?

Reply:
Reasonable, remove it.